### DISCUSSION TITLE
Update price display on an attribute change: add Dynamic Price Updater functionality

### DISCUSSION THREAD

**Discussion by torvista** [zencart/zencart#5895] [Ideas]
In my opinion the functionality of displaying/updating a price based on the current selection is in 2023 something that should be a core function.
Instead there is only Dynamic Price Updater which is extremely complex to maintain and is now spread over four forks thanks to well-meaning efforts.

https://www.zen-cart.com/showthread.php?70577-Dynamic-Price-Updater&p=1396083#post1396083

I gave up/could not justify the time once I tried to understand the vagaries of handling the various discount pricing options.
I think the big guns need to be aimed at it, or a core solution built.

**Comment by Goldbuckl** [comment]
Price updating based on input into the quantity box and attribute selection should be a standard part of the core files.  Other popular ecommerce platforms have integrated this functionality years ago as it takes a cart package to the next level in terms of offering a professional web presence.  No customer wants to fumble with trying to figure out how much something is going to cost them and it is far easier for them to abandon your store and find another where there is no fumbling through to get an instant price.

I started using ZenCart about 20 years ago and at the time was extremely popular.  Doesn't seem as popular now.  My #1 reason for choosing ZenCart at the time was for SEO which is where it shines for small businesses who need a web presence.

My biggest complaint way back then was adding and pricing products that was quantity and attribute based into ZC's cookie cutter type 1 size fits all approach for products.  Then came the DPU Price Updater module and it opened new opportunities for me to show the end user/customer a price to include setup fee(s) (one time charges), running charges as unit prices (great for imprinting), and multiple options/attributes to configure a "custom" product.  The problem with the DPU module is it seems to be poorly maintained with each new ZenCart update and unfortunately its above my paygrade to fix.  I strongly urge the developers of ZenCart to add this functionality to the core files in v2.0.0 as this will once again make ZC a more powerful option and bring ZC to the forefront against other cart platforms particularly open source ecommerce options.

Thanks,
Goldbuckle

**Comment by torvista** [comment]
>The problem with the DPU module is it seems to be poorly maintained  

I would say that due to it's complexity, it has only gotten attention when a customer of a developer flags a problem. So four forks have evolved. I tried for days and gave up.

While of course I agree that the functionality should be core, I'm not sure that this add-on should be the basis...literally: I don't know if there is a better plan/more robust design of code it to make it more maintainable.
Hence my comment above that this needs a decision by a maintainer if it is of core interest and if so, what should be the plan, and then for a volunteer to dedicate a LOT of time to it.

Off-topic, there are several complex PRs in the shadows that people have come up with, obviously investing a huge amount of time and skill, but for which it is debatable that they are on any wishlist or solve any real problem: they are just things that someone felt like doing in their little spare time.
Such ad-hoc/inefficient development is the result of the lack of any development roadmap as guidance/suggestions for development, something of which I have long-complained.

With a roadmap, perhaps the functionality desired in this thread would have been long included.
